### PR TITLE
PAR - Thibaut Davoult - done 1st half (celebrities, not movies)

### PR DESCRIPTION
**Pull request by thibautdavoult**
wasted 4 hours on a typo in seeds.js

### CODE REVIEW

**Comment by thibautdavoult** [comment]
On my router celebrities.js, line 66: `Celebrity.update({_id: celebId}, updatedCelebInfo, (err, celebrity) => {`

Not sure why using `celebId` as first argument didn't work (it was updating not the current celebrity but the first on the list, all the time). Putting it into an object worked, which isn't what's shown on the lesson: http://materials.ironhack.com/s/ryYU6ZVDg#save-product-to-database

So I'm a bit confused about that!

File: movies/routes/celebrities.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+
+router.post("/celebrities/:id", (req, res, next) => {
+  const celebId = req.params.id;
+  const updatedCelebInfo = {
+    name: req.body.name,
+    occupation: req.body.occupation,
+    catchPhrase: req.body.catchPhrase,
+  };
+  Celebrity.update({_id: celebId}, updatedCelebInfo, (err, celebrity) => {
```
Comment (mlande) [inline movies/routes/celebrities.js]: The reason you need to pass an object as an argument here, is that you are using the mongoose Model.update method, which "Updates all documents matching conditions using the update clause." as such:

var conditions = { name: 'bourne' }
  , update = { $inc: { visits: 1 }}
  , options = { multi: true };
Model.update(conditions, update, options, callback);

see http://mongoosejs.com/docs/2.7.x/docs/updating-documents.html

This method expects an object with a list of properties to match, while the method used in the learning unit is a similar but different Model.findByIdAndUpdate which expects an obejct id as argument. Always make sure you are passing the query what it expects :)

File: movies/views/celebrities/index.ejs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+<h2>List of Celebs</h2>
+
+<ul>
+  <% celebrities.forEach((celeb) => { %>
+    <li>
+      <a href="/celebrities/<%= celeb.id %>">Name: <%= celeb.name %></a>
+      <form style="display: inline;" action="/celebrities/<%= celeb.id %>/delete" method="POST"><input type="submit"  value="Delete"></form>
+      <form style="display: inline;" action="/celebrities/<%= celeb.id %>/edit" method="GET"><input type="submit"  value="Edit"></form>
+    </li>
+
```
Comment (mlande) [inline movies/views/celebrities/index.ejs]: This view is good, but it lacks a link so the user can access the celebrities/new view. In this case, I had to manually type the url in the browser to be able to add one. Think about a user accessing the page the first time and what he would want to do :)`

**Comment by mlande** [comment]
Good work on the API, seems that you got the hang of routing on the server side. Now just make sure that the functionality yu created on the server, such as adding a celeb, is also easily accessible to the user. See my comments for more info.

I also answered your question about the query in the comments (hopefully :)

Keep up the good work

File: movies/views/celebrities/index.ejs (hunk 2 above)
Comment (thibautdavoult) [reply movies/views/celebrities/index.ejs]: Big oversight on my part. I'll definitely update that while doing the movies part of the exercise!

File: movies/routes/celebrities.js (hunk 1 above)
Comment (thibautdavoult) [reply movies/routes/celebrities.js]: That makes a lot of sense, I just didin't notice that we used 2 different methods. Thanks a lot for the clarification!

**Pull request closed without merging**